src/rust/inetstack/protocols/tcp/queue.rs

```diff
@@ -43,8 +49,11 @@ impl<const N: usize> TcpQueue<N> {
     }
 
     /// Set underlying TCP socket data structure.
-    pub fn set_socket(&mut self, s: Socket<N>) {
-        self.socket = s;
+    pub fn set_socket(&mut self, mut s: Socket<N>) {
+        swap(&mut s, &mut self.socket);
```

Comment:
What's the reason for using swap here?

Reply:
If I understood correctly, he is doing this because we explicitly want to skip the destructor on the next statement.